File: E-Tour/src/main/java/com/example/demo/entities/Ads.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "ads")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Ads {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ad_id")
    @EqualsAndHashCode.Include
    private Integer adId;

    @Column(name = "title", length = 150)
    private String title;

    @Column(name = "image_path", length = 255)
    private String imagePath;

    @Column(name = "link_url", length = 255)
    private String linkUrl;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "display_order")
    private Integer displayOrder;

    @Column(name = "active")
    private Boolean active;
}

File: ETour-DotNet/Models/Ads.cs
namespace ETour.Api.Models;

public class Ads
{
    public int AdId { get; set; }
    public string Title { get; set; }
    public string ImagePath { get; set; }
    public string LinkUrl { get; set; }
    public DateOnly? StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public int? DisplayOrder { get; set; }
    public bool? Active { get; set; }
}
